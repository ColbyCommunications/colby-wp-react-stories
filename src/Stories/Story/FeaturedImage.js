import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledImage = styled.img`
  width: 100%;
  height: auto;
`;

const FeaturedImage = ({ altText, featuredImage }) => {
  try {
    const data =
      featuredImage.media_details.sizes.large ||
      featuredImage.media_details.sizes.medium;

    return (
      <StyledImage
        alt={altText}
        src={data.source_url}
        width={data.width}
        height={data.height}
      />
    );
  } catch (e) {
    return null;
  }
};

FeaturedImage.propTypes = {
  altText: PropTypes.string.isRequired,
  featuredImage: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default FeaturedImage;
