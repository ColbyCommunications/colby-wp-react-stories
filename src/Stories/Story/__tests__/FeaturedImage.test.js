import React from 'react';
import renderer from 'react-test-renderer';

import FeaturedImage from '../FeaturedImage';
import { media } from '../../__tests__/testData';

describe('FeaturedImage', () => {
  it('handles an invalid image object', () => {
    const tree = renderer.create(<FeaturedImage
      featuredImage={{}}
      altText="My alt text"
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('handles an image object with a large photo', () => {
    const tree = renderer.create(<FeaturedImage
      featuredImage={media}
      altText="My alt text"
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('handles an image object with a medium photo', () => {
    const featuredImage = { ...media };
    featuredImage.media_details.sizes.medium = { ...featuredImage.media_details.sizes.large };
    delete featuredImage.media_details.sizes.large;

    const tree = renderer.create(<FeaturedImage
      featuredImage={featuredImage}
      altText="My alt text"
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
