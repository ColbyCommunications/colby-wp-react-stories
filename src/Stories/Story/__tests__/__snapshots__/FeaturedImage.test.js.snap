// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedImage handles an image object with a large photo 1`] = `
<img
  alt="My alt text"
  className="sc-bdVaJa ctpUMM"
  height={100}
  src="http://my-image"
  width={200}
/>
`;

exports[`FeaturedImage handles an image object with a medium photo 1`] = `
<img
  alt="My alt text"
  className="sc-bdVaJa ctpUMM"
  height={100}
  src="http://my-image"
  width={200}
/>
`;

exports[`FeaturedImage handles an invalid image object 1`] = `null`;
